qiita_db/study.py

```diff
@@ -673,7 +674,13 @@ def sample_template(self):
         -------
         SampleTemplate id
         """
-        return self._id
+        with TRN:
+            sql = """SELECT EXISTS(SELECT *
+                                   FROM qiita.study_sample
+                                   WHERE study_id = %s)"""
+            TRN.add(sql, [self.id])
+            exists = TRN.execute_fetchlast()
+        return self._id if exists else None
```

Comment:
I've mixed fillings here... If we use `-1` if there is any part of the code that assumes that the Sample Template exists and it actually doesn't, it can generate some errors that may be obscure to find. Raising an error has the advantage that we will quickly identify the source of the error.

On the other hand, if we want to check if a study has a sample template or not, we will be able to check if the returned id is `-1`, while with the error we will need to have a try/except.

@antgonza @ElDeveloper after these thoughts, which one do you think is the best one moving forward?
If I have to choose, I'm slightly towards returning -1, but not sure.